.github/workflows/integration.yml

```diff
@@ -31,27 +31,17 @@ jobs:
       - name: Lint
         run: python3 -m pylint --rcfile=.pylintrc *.py **/*.py
 
-      - name: Unit Test
-        run: ORIGINS=testorigin python3 -m pytest -o log_cli=true --disable-warnings --verbose
+      - name: Unit Test (with coverage)
+        run: ORIGINS=testorigin coverage run --source=. -m pytest -o log_cli=true --disable-warnings --verbose
 
-      - name: Run coverage and SonarScanner
-        # - Re-run unit test with coverage.
-        # - Install sonar-scanner.
-        # - Run sonar-scanner.
+      - name: Create coverage report
         shell: bash
-        env:
-          SONARCLOUD_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
         run: |
-          echo "---Install sonar-scanner---"
-          export SONAR_SCANNER_VERSION=4.2.0.1873
-          export SONAR_SCANNER_HOME=$HOME/.sonar/sonar-scanner-$SONAR_SCANNER_VERSION-linux
-          curl --create-dirs -sSLo $HOME/.sonar/sonar-scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-$SONAR_SCANNER_VERSION-linux.zip
-          unzip -o $HOME/.sonar/sonar-scanner.zip -d $HOME/.sonar/
-          export PATH=$SONAR_SCANNER_HOME/bin:$PATH
-          export SONAR_SCANNER_OPTS="-server"
-          echo "---Run tests with coverage---"
-          ORIGINS=testorigin coverage run --source=. -m pytest
           coverage report
           coverage xml -o coverage-reports/coverage-report.xml
```

Comment:
what happens to this file `coverage-report.xml`? it gets uploaded to Sonarcloud or something?

Reply:
Yep - sonar-runner pushes it to Sonarcloud.